python/unitytrainers/ppo/models.py

```diff
@@ -44,6 +49,92 @@ def create_reward_encoder():
+                encoded_visual = self.create_visual_observation_encoder(self.visual_in[i], 128,
+                                                                        self.swish, 1, "visual_obs_encoder", False)
+                encoded_next_visual = self.create_visual_observation_encoder(self.next_visual_in[i], 128,
+                                                                             self.swish, 1, "visual_obs_encoder", True)
+                visual_encoders.append(encoded_visual)
+                next_visual_encoders.append(encoded_next_visual)
+
+            hidden_visual = tf.concat(visual_encoders, axis=1)
+            hidden_next_visual = tf.concat(next_visual_encoders, axis=1)
+            inverse_input_list.append(hidden_visual)
+            inverse_input_list.append(hidden_next_visual)
+            encoded_state_list.append(hidden_visual)
+            encoded_next_state_list.append(hidden_next_visual)
+
+        if self.o_size > 0:
+            # Create input op for next (t+1) vector observation.
+            self.next_vector_obs = tf.placeholder(shape=[None, self.o_size], dtype=tf.float32,
+                                                  name='next_vector_observation')
+
+            # Create the encoder ops for current and next vector input. Not that these encoders are siamese.
+            encoded_vector_obs = self.create_continuous_observation_encoder(self.vector_in, 128, self.swish, 2,
+                                                                            "vector_obs_encoder", False)
+            encoded_next_vector_obs = self.create_continuous_observation_encoder(self.next_vector_obs, 128, self.swish,
+                                                                                 2, "vector_obs_encoder", True)
+
+            inverse_input_list.append(encoded_vector_obs)
+            inverse_input_list.append(encoded_next_vector_obs)
+            encoded_state_list.append(encoded_vector_obs)
+            encoded_next_state_list.append(encoded_next_vector_obs)
+
+        encoded_state = tf.concat(encoded_state_list, axis=1)
+        encoded_next_state = tf.concat(encoded_next_state_list, axis=1)
+        return encoded_state, encoded_next_state
+
+    def create_inverse_model(self, encoded_state, encoded_next_state):
+        """
+        Creates inverse model TensorFlow ops for Curiosity module.
+        :param encoded_state: Tensor corresponding to encoded current state.
+        :param encoded_next_state: Tensor corresponding to encoded next state.
+        """
+        combined_input = tf.concat([encoded_state, encoded_next_state], axis=1)
+        if self.brain.vector_action_space_type == "continuous":
+            pred_action = tf.layers.dense(combined_input, self.a_size, activation=None)
+            squared_difference = tf.reduce_sum(tf.squared_difference(pred_action, self.selected_actions), axis=1)
```

Comment:
Where is self.selected_actions created in continuous control. In discrete, it is in create_dc_actor_critic. I think self.selected_actions should be in create_dc_actor_critic then. Did you try curiosity with a continuous control environment to make sure self.selected_action is used ?